source/gui/settingsDialogs.py

```diff
@@ -5445,3 +5526,196 @@ def onFilterEditTextChange(self, evt):
 		self.filter(self.filterEdit.Value)
 		self._refreshVisibleItems()
 		evt.Skip()
+
+
+class SetURLDialog(SettingsDialog):
```

Comment:
> I think it also might be worth moving this to a separate file if possible.

I think that might cause problems with circular imports, since it's based on `gui.settingsDialogs.SettingsDialog`, and is only used in `gui.settingsDialogs`.

> is this intended to be used by the add-on mirror URL too?

That's the idea, yes. I thought it best to implement this here, and then open a new PR adding these new controls to the Add-on Store settings/

Reply:
I think you'd be able to lazily import `SetURLDialog` into wherever it's needed in `gui.settingsDialogs`, so it shouldn't cause a circular import. I think it's important we try to split up `gui.settingsDialogs` and moving more versatile/helper dialogs is a good start.